### PR TITLE
Support composer install

### PR DESCRIPTION
**Pull request by riaf**
I've added support [Composer](http://getcomposer.org).
However, I do not have experience using HHVM. Is this way right?

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -44,3 +45,22 @@ test -d "vendor/hhvm"
...
+  if [ ! -f "composer.phar" ]; then
+    cout "Fetching composer.phar"
+    curl -LOs $COMPOSER_URL
+  fi
+
+  cout "Running: composer.phar install"
+
+  GIT_DIR_ORIG=$GIT_DIR
+  unset GIT_DIR
+  LD_LIBRARY_PATH=vendor/hhvm/ vendor/hhvm/hhvm -c vendor/hhvm/config.hdf composer.phar install -n --prefer-source | indent
```
Comment (henrikbjorn) [inline bin/compile:61]: `--prefer-source` is wrong and should be removed. HHVM does not support Zip Extension but composer falls back to unzip command line (not sure if that is useable on heroku).

Comment (riaf) [inline bin/compile:61]: Heroku does not have unzip, are there other methods to solve it?

**Comment by ptarjan** [comment]
I like this, but would like @henrikbjorn to weigh in on the zip issue before I merge. Thanks for the PR.

**Comment by henrikbjorn** [comment]
I think that the only way to know for sure if unzip system command is available when installing with composer is to try it out.

**Comment by ptarjan** [comment]
```
$ heroku run bash
Running `bash` attached to terminal... up, run.9523
~ $ unzip
bash: unzip: command not found
~ $ 
```

**Comment by ptarjan** [comment]
@henrikbjorn is there any harm to `prefer-source?`

**Comment by henrikbjorn** [comment]
@ptarjan only that it takes a lot more resources to clone for git, hg and you will get huge folders (as git contains everything).

**Comment by henrikbjorn** [comment]
Seems like that ZipExtension have landed in HHVM master, maybe this should be reopened and reevaluated.

**Comment by ptarjan** [comment]
@henrikbjorn we will have to wait for the next binary release and then we can try using zip in composer then

**Comment by henrikbjorn** [comment]
Is there a release cycle?

**Comment by ptarjan** [comment]
@henrikbjorn no formal time on the clock, but whenever a large change lands we will release a point release. One is coming in the next week for fastcgi.

**Comment by nubs** [comment]
It should be possible to remove `--prefer-source` now correct?

**Comment by ptarjan** [comment]
@nubs https://github.com/hhvm/heroku-buildpack-hhvm/commit/1da769bc26143a81182535f0f540ac71cf7ee48e try it out and open an issue if it doens't work for you

**Pull request merged**